Description: Remove non-required static lib dependency
 Build should not require -lfollylogging and -lfollyinit when statically
 linked
 Also, static library ordering was causing undefined references, needed to
 remove wangle from the autoconf deps and add it in LDADD flags.
 .
 mcrouter (0.38.0-2) xenial; urgency=medium
 .
   * Version 0.38.0
     Removed non-required static lib dependency on libfollyinit,
     libfollylogging
Author: Ori Shalev <orish@fb.com>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: https://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- mcrouter-0.38.0.orig/mcrouter/Makefile.am
+++ mcrouter-0.38.0/mcrouter/Makefile.am
@@ -200,7 +200,7 @@ mcrouter_SOURCES = \
 
 libmcroutercore_a_CPPFLAGS = -I..
 
-mcrouter_LDADD = libmcroutercore.a lib/libmcrouter.a -lfolly -lfollylogging
+mcrouter_LDADD = libmcroutercore.a lib/libmcrouter.a -lwangle -lfolly
 mcrouter_CPPFLAGS = -I..
 
 clean-local:
--- mcrouter-0.38.0.orig/mcrouter/configure.ac
+++ mcrouter-0.38.0/mcrouter/configure.ac
@@ -112,13 +112,13 @@ AC_CHECK_LIB([z], [gzread], [], [AC_MSG_
 AC_CHECK_LIB([double-conversion],[ceil],[],[AC_MSG_ERROR(
              [Please install double-conversion library])])
 AC_CHECK_LIB([dl], [dlopen], [])
-AC_CHECK_LIB([folly],[follyinit],[follylogging],[getenv],[],[AC_MSG_ERROR(
+AC_CHECK_LIB([folly],[getenv],[],[AC_MSG_ERROR(
              [Please install the folly library])])
 AC_CHECK_HEADER([folly/Likely.h], [], [AC_MSG_ERROR(
 [Couldn't find folly, please download from https://github.com/facebook/folly]
 )], [])
-AC_CHECK_LIB([wangle], [getenv], [], [AC_MSG_ERROR(
-             [Please install the wangle library])])
+#AC_CHECK_LIB([wangle], [getenv], [], [AC_MSG_ERROR(
+#             [Please install the wangle library])])
 
 # Checks for typedefs, structures, and compiler characteristics.
 AC_HEADER_STDBOOL
@@ -155,7 +155,7 @@ AC_CHECK_FUNCS([gettimeofday \
 
 LIBS="$LIBS $BOOST_LDFLAGS $BOOST_CONTEXT_LIB $BOOST_FILESYSTEM_LIB \
       $BOOST_PROGRAM_OPTIONS_LIB $BOOST_SYSTEM_LIB $BOOST_REGEX_LIB \
-      $BOOST_THREAD_LIB -lpthread -pthread -ldl"
+      $BOOST_THREAD_LIB -lpthread -pthread"
 
 AM_PATH_PYTHON([2.6],, [:])
 AM_CONDITIONAL([HAVE_PYTHON], [test "$PYTHON" != :])
--- mcrouter-0.38.0.orig/mcrouter/lib/config/test/Makefile.am
+++ mcrouter-0.38.0/mcrouter/lib/config/test/Makefile.am
@@ -11,6 +11,4 @@ mcrouter_config_test_CPPFLAGS = \
 
 mcrouter_config_test_LDADD = \
   $(top_builddir)/lib/libmcrouter.a \
-  $(top_builddir)/lib/libtestmain.la \
-	-lfollylogging \
-	-lfollyinit
+  $(top_builddir)/lib/libtestmain.la
--- mcrouter-0.38.0.orig/mcrouter/lib/fbi/cpp/test/Makefile.am
+++ mcrouter-0.38.0/mcrouter/lib/fbi/cpp/test/Makefile.am
@@ -10,6 +10,4 @@ mcrouter_fbi_cpp_test_CPPFLAGS = \
 
 mcrouter_fbi_cpp_test_LDADD = \
   $(top_builddir)/lib/libmcrouter.a \
-  $(top_builddir)/lib/libtestmain.la \
-	-lfollylogging \
-	-lfollyinit
+  $(top_builddir)/lib/libtestmain.la
--- mcrouter-0.38.0.orig/mcrouter/lib/fbi/test/Makefile.am
+++ mcrouter-0.38.0/mcrouter/lib/fbi/test/Makefile.am
@@ -14,6 +14,4 @@ mcrouter_fbi_test_CPPFLAGS = \
 
 mcrouter_fbi_test_LDADD = \
 	$(top_builddir)/lib/libmcrouter.a \
-	$(top_builddir)/lib/libtestmain.la \
-	-lfollylogging \
-	-lfollyinit
+	$(top_builddir)/lib/libtestmain.la
--- mcrouter-0.38.0.orig/mcrouter/lib/network/Makefile.am
+++ mcrouter-0.38.0/mcrouter/lib/network/Makefile.am
@@ -10,11 +10,7 @@ mock_mc_server_SOURCES = \
 
 mock_mc_server_CPPFLAGS = -I$(top_srcdir)/..
 
-mock_mc_server_LDADD = \
-	$(top_builddir)/lib/libmcrouter.a \
-	-lfolly \
-	-lfollyinit \
-	-lfollylogging
+mock_mc_server_LDADD = $(top_builddir)/lib/libmcrouter.a -lwangle -lfolly
 
 libtest_util_a_SOURCES = \
   test/ClientSocket.cpp \
--- mcrouter-0.38.0.orig/mcrouter/lib/network/test/Makefile.am
+++ mcrouter-0.38.0/mcrouter/lib/network/test/Makefile.am
@@ -26,6 +26,4 @@ mcrouter_network_test_CPPFLAGS = \
 mcrouter_network_test_LDADD = \
   $(top_builddir)/lib/libmcrouter.a \
   $(top_builddir)/lib/libtestmain.la \
-  $(top_builddir)/lib/network/libtest_util.a \
-	-lfollylogging \
-	-lfollyinit
+  $(top_builddir)/lib/network/libtest_util.a
--- mcrouter-0.38.0.orig/mcrouter/lib/test/Makefile.am
+++ mcrouter-0.38.0/mcrouter/lib/test/Makefile.am
@@ -22,6 +22,4 @@ mcrouter_lib_test_CPPFLAGS = \
 
 mcrouter_lib_test_LDADD = \
   $(top_builddir)/lib/libmcrouter.a \
-  $(top_builddir)/lib/libtestmain.la \
-	-lfollylogging \
-	-lfollyinit
+  $(top_builddir)/lib/libtestmain.la
--- mcrouter-0.38.0.orig/mcrouter/routes/test/Makefile.am
+++ mcrouter-0.38.0/mcrouter/routes/test/Makefile.am
@@ -20,6 +20,4 @@ mcrouter_routes_test_LDADD = \
   $(top_builddir)/libmcroutercore.a \
   $(top_builddir)/lib/libmcrouter.a \
   $(top_builddir)/lib/libtestmain.la \
-  -lfollybenchmark \
-  -lfollyinit \
-  -lfollylogging
+  -lfollybenchmark
--- mcrouter-0.38.0.orig/mcrouter/test/cpp_unit_tests/Makefile.am
+++ mcrouter-0.38.0/mcrouter/test/cpp_unit_tests/Makefile.am
@@ -25,6 +25,4 @@ mcrouter_test_LDADD = \
   $(top_builddir)/libmcroutercore.a \
   $(top_builddir)/lib/libmcrouter.a \
   $(top_builddir)/lib/libtestmain.la \
-  $(top_builddir)/lib/network/libtest_util.a \
-	-lfollylogging \
-	-lfollyinit
+  $(top_builddir)/lib/network/libtest_util.a
--- mcrouter-0.38.0.orig/mcrouter/tools/mcpiper/Makefile.am
+++ mcrouter-0.38.0/mcrouter/tools/mcpiper/Makefile.am
@@ -31,8 +31,6 @@ mcpiper_SOURCES = \
 
 mcpiper_LDADD = \
 	$(top_srcdir)/lib/libmcrouter.a \
-	-lfolly \
-	-lfollylogging \
-	-lfollyinit
+	-lfolly
 
 mcpiper_CPPFLAGS = -I$(top_srcdir)/..
